jlm/rvsdg/node.hpp

```diff
@@ -243,12 +247,237 @@ class Output
   friend class rvsdg::Region;
 
 public:
-  using UserIterator = util::PtrIterator<Input, std::unordered_set<Input *>::iterator>;
-  using UserConstIterator =
-      util::PtrIterator<const Input, std::unordered_set<Input *>::const_iterator>;
+  class UserIterator
```

Comment:
It was not possible to reuse [Input::Iterator](https://github.com/phate/jlm/blob/cf502f1b85c780970c00cf4c3537a7f069402e08/jlm/rvsdg/node.hpp#L83)?

Reply:
changed it

Reply:
sorry, did not change, but Input::Iterator does something else -- as discussed should move somewhere else